### PR TITLE
Resolve links for strongly typed models

### LINKED ISSUE #18: Strongly typed models should have all rich_text links resolved
At least I think :) 

Right now the value is simply copied.

2017-03-16 15_34_33-settings

### PR DESCRIPTION
**Pull request by ondrejsevcik**
This should fix #18 

Hopefully I got it right :) 

Before it's merged, we should probably store content items used in this test in Test project rather than in my sandbox project.

### CODE REVIEW

File: KenticoCloud.Delivery.Tests/ContentLinkResolverTests.cs [hunk 1]
```diff
@@ -107,6 +107,22 @@ public void ContentLinkAttributesAreParsed()
...
+
+        [Test]
+        public void ResolveLinksInStronglyTypedModel()
+        {
+            // TODO: Use test project rather than sandbox
+            var client = new DeliveryClient("e1167a11-75af-4a08-ad84-0582b463b010");
+            client.ContentLinkUrlResolver = new CustomContentLinkUrlResolver();
+
+            var item = client.GetItemAsync<Office>("brno_office").Result.Item;
+            Assert.AreEqual("", item.AboutTheOffce);
```
Comment (petrsvihlik) [inline KenticoCloud.Delivery.Tests/ContentLinkResolverTests.cs]: this is failing...you probably shouldn't compare it with an empty text

File: KenticoCloud.Delivery/CodeFirst/CodeFirstModelProvider.cs [hunk 2]
```diff
@@ -84,19 +84,29 @@ internal object GetContentItemModel(Type t, JToken item, JToken modularContent,
...
 
-                        if (propertyType == typeof(IEnumerable<MultipleChoiceOption>)
+                        if (propertyType == typeof(string))
+                        {
+                            var links = ((JObject)propValue?.Parent?.Parent)?.Property("links")?.Value;
+
+                            // Handle rich_text link resolution
+                            if (links != null && propValue != null && _client.ContentLinkResolver != null)
+                            {
+                                value = _client.ContentLinkResolver.ResolveContentLinks(propValue?.ToObject<string>(), links);
```
Comment (petrsvihlik) [inline KenticoCloud.Delivery/CodeFirst/CodeFirstModelProvider.cs]: the resolved value is never used

File: KenticoCloud.Delivery/CodeFirst/CodeFirstModelProvider.cs [hunk 3]
```diff
@@ -84,19 +84,29 @@ internal object GetContentItemModel(Type t, JToken item, JToken modularContent,
...
+                            }
+
+                            value = propValue?.ToObject(propertyType);
```
Comment (petrsvihlik) [inline KenticoCloud.Delivery/CodeFirst/CodeFirstModelProvider.cs]: the resolved value gets overwritten with this statement

**Comment by petrsvihlik** [comment]
we'll address the inconsistency in project ids used in tests separately.

**Comment by ondrejsevcik** [comment]
omg, sorry about that! It should be fixed by now.

**Pull request merged**